dspace-ldn/src/main/java/org/dspace/app/ldn/processor/LDNMetadataProcessor.java

```diff
@@ -314,6 +334,27 @@ private Item lookupItem(Context context, Notification notification) throws SQLEx
         return item;
     }
 
+    /**
+     * Resolves context id externally if not already a DSpace URL. Context id must
+     * resolve as a 302 with Location header.
+     *
+     * @param notification current context notification
+     * @return external resolved DSpace URL
+     */
+    private String resolveContext(Notification notification) {
+        String url = notification.getContext().getId();
+        if (!url.startsWith(this.dspaceUIUrl)) {
+            log.info("Resolving external context id {}", url);
+            HttpHeaders headers = this.restTemplate.headForHeaders(url);
+            if (headers.containsKey(LOCATION_HEADER_KEY)) {
+                return headers.getFirst(LOCATION_HEADER_KEY);
+            } else {
+                log.error("External context id {} HEAD response did not contain Location header", url);
```

Comment:
Is this always an `error`? I thought the scenario could also exist of this URL pointing to another repository (like [arxiv.org](https://arxiv.org/))... or would that scenario not get to this point? If this is an error, is logging it sufficient?

Reply:
The context id of a notification must be able lookup an item in DSpace. The two options are UUID or handle, which the notification processor supports both being extracted from a DSpace URL. We are introducing an option for an external resolver to supply the DSpace UI URL. I am not sure of how it could possible work making a head request to another repository in which does not return a Location header containing a link to a DSpace item. Well at least how it could work with generalized logic.

My original assumption for only logging an error was that the item lookup would inevitably fail and throw the appropriate response. Thinking a bit more about it, it makes sense to throw an exception here an avoid performing an item lookup in which we know will already fail. I will add an appropriate exception throw in addition to the error log.

Reply:
Then again, if the location header exists and is not a valid item DSpace URL, it will fail item lookup all the same.

Reply:
In the case in which the context id is not a DSpace URL and a HEAD request does not return a Location header, what should be the response exception of attempt to send notification? I am thinking 400 bad request.